File: src/main/java/com/github/fenixsoft/bookstore/applicaiton/billnote/UserService.java
package com.github.fenixsoft.bookstore.applicaiton.billnote;

import com.github.fenixsoft.bookstore.domain.billnote.entity.User;
import com.github.fenixsoft.bookstore.infrastructure.billnote.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public User registerUser(String username, String email, String password) {
        User user = new User();
        user.setUsername(username);
        user.setEmail(email);
        user.setPasswordHash(password);
        return userRepository.save(user);
    }

    public Optional<User> findByUsername(String username) {
        return userRepository.findByUsername(username);
    }

    public void deleteUser(Long userId) {
        userRepository.deleteById(userId);
    }

    public User updateUser(Long userId, String newEmail, String newPassword) {
        User user = userRepository.findById(userId).orElseThrow(() -> new RuntimeException("User not found"));
        user.setEmail(newEmail);
        user.setPasswordHash(newPassword);
        return userRepository.save(user);
    }
}

File: frontend/src/services/userService.js
import http from './http'

class UserService {
  // 获取所有用户
  getUsers() {
    return http.get('/users')
  }

  // 获取用户详情
  getUser(id) {
    return http.get(`/users/${id}`)
  }

  // 创建用户
  createUser(user) {
    return http.post('/users', user)
  }

  // 更新用户
  updateUser(id, user) {
    return http.put(`/users/${id}`, user)
  }

  // 删除用户
  deleteUser(id) {
    return http.delete(`/users/${id}`)
  }
}

export default new UserService()